src/components/Nav/Navigation.tsx

```diff
@@ -77,16 +76,11 @@ class Navigation extends React.Component<PropsType, StateType> {
     }
   };
 
-  navigateTo(e: any) {
-    if (e.title === servicesTitle) {
-      this.context.router.history.push(servicesPath);
-    } else if (e.title === istioRulesTitle) {
-      this.context.router.history.push(istioRulesPath);
-    } else if (e.title === servicesJaeger) {
-      this.context.router.history.push(servicesJaegerPath);
-    } else {
-      this.context.router.history.push(serviceGraphPath);
-    }
+  componentWillReceiveProps(nextProps: PropsType) {
```

Comment:
This method seems deprecated/unsafe [1], please use ´getDerivedStateFromProps´ [2]


[1] https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops
[2] https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
Oh... I'll fix that...
So... I need to re-read the docs.

Reply:
I suppose it will suffice to read from here https://github.com/kiali/kiali-ui/pull/214#issue-179554533 (the comments and the links)